### ISSUE TITLE
multiline-ternary - allow "switch-like" style ("true" branch on the same line as condition), also specify colon location

### ISSUE THREAD

**Issue by KillyMXI** [eslint/eslint#13448]
**What rule do you want to change?**

[multiline-ternary](https://eslint.org/docs/rules/multiline-ternary)

**Does this change cause the rule to produce more or fewer warnings?**

It depends. The point is to have more granular control over the style.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New options. I'm not sure about the names though.

- I think it makes sense to have a list to specify acceptable multiline styles (need help to come up with good short names though, especially for currently allowed style).
- And an option for colon location (end of line, beginning of line).

**Please provide some example code that this change will affect:**

```js
// following should pass with ["error", "always-multiline", { styles: ["switch-like", ...], colon-location: "beginning" }]

foo > bar ? value1
    : value2;

let x = (foo > bar) ? value1
    : (baz > qux) ? value2
    : (quux > quuz) ? value3
    : value4;

// this should pass with ["error", "always-multiline", { styles: ["switch-like", ...], colon-location: "end" }]

foo > bar ? value1 :
    value2;

let x = (foo > bar) ? value1 :
    (baz > qux) ? value2 :
    (quux > quuz) ? value3 :
    value4;
```

**What does the rule currently do for this code?**

For multiline ternaries there is only one allowed style, where condition and both branches are on their own lines, which is not always most readable.

**What will the rule do after it's changed?**

There should be a combination of options to enforce the style shown above, or have it along with the existing one.

**Are you willing to submit a pull request to implement this change?**

Never wrote an ESLint rule. Maybe, if I get time.

----

Depending on how the discussion will go, colon location can be split into a separate issue. For me it's just wasn't worth it to bring up one without the other.

**Comment by anikethsaha** [comment]
Thanks for the issue, 

`multiline-ternary` is a stylistic rule, so I think it will come under [new policy](https://eslint.org/blog/2020/05/changes-to-rules-policies) under which no new changes to stylistic rules are being accepted for the core rules.

**Comment by KillyMXI** [comment]
Ah.
I wish there was a better community process around plugins then.
Finding a right set of plugins for what I might want is a pain...
If I implement this in my own plugin - I might be missing an already made plugin doing the same thing, and likewise, people after me are unlikely to find my work.

**Comment by anikethsaha** [comment]
Eslint's ecosystem is huge, so it may be difficult to search for specific use-case. 
IMO, you can try to create a rule on top of existing `multiline-ternary` rule using packages like [this](https://github.com/not-an-aardvark/eslint-rule-composer) or approaches like [this](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/brace-style.ts)

**Comment by kaicataldo** [comment]
@KillyMXI FWIW, we would also like to improve the discoverability of plugins. If that's something you'd like to help us solve, we'd love to work with you!
